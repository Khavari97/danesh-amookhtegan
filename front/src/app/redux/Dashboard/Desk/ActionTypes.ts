export const DESK_USER_PROJECTS = "DESK_USER_PROJECTS";
export const DESK_USER_REQUESTS = "DESK_USER_REQUESTS";
export const DESK_LOADING = "DESK_LOADING";

export enum ProjectsStatus {
    WAITING = "WAITING",
    STARTED = "STARTED",
    ENDED = "ENDED",
    DELETED = "DELETED"
}

export interface IProjectCard {
    status: {
        code: ProjectsStatus,
        label: string
    },
    name: string,
    description: string,
    role: string,
    project?: number,
    url: string
}

export enum RequestsStatus {
    PENDING = "PENDING",
    DECLINED = "DECLINED"
}

export interface IRequestedCard {
    status: {
        code: RequestsStatus,
        label: string,
    },
    name: string,
    description: string,
    project?: number,
    url: string
}

export interface DeskState {
    loading: boolean,
    projects: IProjectCard[],
    requests: IRequestedCard[],
}


export interface IUserProjects {
    type: typeof DESK_USER_PROJECTS,
    payload: IProjectCard[],
}

export interface IUserRequests {
    type: typeof DESK_USER_REQUESTS,
    payload: IRequestedCard[],
}

export interface IDeskLoading {
    type: typeof DESK_LOADING,
    payload: boolean
}

export type DeskActionTypes = IUserProjects | IUserRequests | IDeskLoading;