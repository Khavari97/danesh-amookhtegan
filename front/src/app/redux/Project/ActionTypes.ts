export const SET_PROJECT_SLUG = "SET_PROJECT_SLUG"

export interface IProjectState {
    loading: boolean,
    slug: string,
    status?: IProjectStatus,
}

export interface IProjectStatus {
    name: string,
    status: {
        code: string;
        label: string;
    },
    role: {
        code: string;
        label: string;
    }
}

export interface ISetProjectSlug {
    type: typeof SET_PROJECT_SLUG,
    payload: {
        slug: string,
        status: IProjectStatus
    }
}

export type IProjectActions = ISetProjectSlug