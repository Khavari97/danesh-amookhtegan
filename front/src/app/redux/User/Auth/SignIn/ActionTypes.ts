export const REQUEST = 'REQUEST';
export let FETCH_SUCCESS = 'FETCH_SUCCESS';
export let FETCH_FAIL = 'FETCH_FAIL';

export interface IRequest {
    type: typeof REQUEST,
}

export interface ISuccess {
    type: typeof FETCH_SUCCESS,
    payload: string
}

export interface IFail {
    type: typeof FETCH_FAIL,
    payload: string
}


export interface IState {
    loading: boolean,
    res: string,
    error: string
}

export const INITIAL_STATE: IState = {
    loading: true,
    res: "",
    error: ""
}
export type FetchDispatchTypes = IFail | ISuccess | IRequest