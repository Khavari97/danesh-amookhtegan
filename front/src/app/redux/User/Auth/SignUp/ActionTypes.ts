export const SET_PASSWORD = 'SET_PASSWORD';
export const SET_PROFILE = 'SET_PROFILE';

export interface IState {
    password: string,
    profile: object,
}

export interface IPassword {
    type: typeof SET_PASSWORD;
    payload: string;
}

export interface IProfile {
    type: typeof SET_PROFILE;
    payload: object;
}

export const INITIAL_STATE: IState = {
    password: '',
    profile: {},
}

export type SignUpTypes = IPassword | IProfile