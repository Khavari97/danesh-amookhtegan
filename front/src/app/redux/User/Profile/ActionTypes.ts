export const SET_PROVINCES = 'SET_PROVINCES';
export const SET_CITIES = 'SET_CITIES';
export const SET_UNIVERSITY = 'SET_UNIVERSITY'
export const SET_DECRIPTION = 'SET_DECRIPTION'
export const GET_PROFILE = 'GET_PROFILE'
export const GET_PARENT_SKILLS = 'GET_PARENT_SKILLS'
export const GET_SKILLS = 'GET_SKILLS'
export const GET_RESUME = 'GET_RESUME'
export const GET_AVATAR = 'GET_AVATAR'
export const GET_BRIEF_PROFILE = 'GET_BRIEF_PROFILE'

interface University {
    id: number | any
    code: string,
    name: string,
    city: number
    province?: province | any
}

interface Universities extends Array<University> {
}

export interface IUniversities {
    type: typeof SET_UNIVERSITY,
    payload: Universities | any
}

interface City {
    id: number | any
    code: string,
    name: string,
    province?: province | any
}

interface Cities extends Array<City> {
}

export interface ICities {
    type: typeof SET_CITIES,
    payload: Cities | any
}

interface province {
    id: number | any
    code: string,
    name: string
}

interface provinces extends Array<province> {
}

export interface IProvinces {
    type: typeof SET_PROVINCES,
    payload: provinces | any
}


export interface IProfile {
    type: typeof SET_PROVINCES,
    payload: provinces | any
}

export interface IDescription {
    type: typeof SET_DECRIPTION,
    payload: string
}

//profile get
export interface IGetProfile {
    loading: boolean,
    type: typeof GET_PROFILE,
    description: string | any
    data: any
}

export interface IGET_BRIEF_PROFILE {
    type: typeof GET_BRIEF_PROFILE,
    briefprofile: any
    avatar: any
}

export interface IGetResume {
    type: typeof GET_RESUME,
    resume: string
}

export interface IGetAvatar {
    type: typeof GET_AVATAR,
    avatar: string
}

export interface ParentSkill {
    code: string
    name: string
    skill: any
    image: string | any
    isAdd?: boolean
}

interface ParentSkills extends Array<ParentSkill> {
}

export interface IGetParentSkills {
    type: typeof GET_PARENT_SKILLS,
    parentSkills: ParentSkills
}

export interface Skill {
    id: number
    name: string
}

interface Skills extends Array<Skill> {
}

export interface IGetSkills {
    type: typeof GET_SKILLS,
    skills: Skills
}

export interface IState {
    provinces: provinces | any
    profile: object
    cities: Cities
    universities: Universities
    description: string | any
    data: object | any
    parentSkills: ParentSkills
    skills: Skills
    loading: boolean
    resume: string | null
    avatar: any
    briefprofile: any
}

export const INITIAL_STATE: IState = {
    loading: true,
    provinces: [],
    cities: [],
    universities: [],
    profile: {},
    description: '',
    data: {},
    parentSkills: [],
    skills: [],
    resume: null,
    avatar: null,
    briefprofile: {}
}

export type ProfileTypes = IProvinces | ICities | IUniversities | IDescription |
    IGetProfile | IGetParentSkills | IGetSkills | IGetResume | IGetAvatar | IGET_BRIEF_PROFILE
