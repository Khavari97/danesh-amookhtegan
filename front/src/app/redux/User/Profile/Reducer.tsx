import {
    GET_AVATAR,
    GET_BRIEF_PROFILE,
    GET_PARENT_SKILLS,
    GET_PROFILE,
    GET_RESUME,
    GET_SKILLS,
    INITIAL_STATE,
    IState,
    ProfileTypes,
    SET_CITIES,
    SET_DECRIPTION,
    SET_PROVINCES,
    SET_UNIVERSITY
} from "./ActionTypes";


const ReducerProfile = (state = INITIAL_STATE, action: ProfileTypes): IState => {
    switch (action.type) {
        case SET_PROVINCES: {
            return {
                ...state,
                provinces: action.payload
            }
        }
        case SET_CITIES: {
            return {
                ...state,
                cities: action.payload
            }
        }
        case SET_UNIVERSITY: {
            return {
                ...state,
                universities: action.payload
            }
        }
        case SET_DECRIPTION: {
            return {
                ...state,
                description: action.payload
            }
        }
        case GET_PROFILE: {
            return {
                ...state,
                data: action.data,
                description: action.description,
                loading: false
            }
        }
        case GET_PARENT_SKILLS: {
            return {
                ...state,
                parentSkills: action.parentSkills,
            }
        }
        case GET_SKILLS: {
            return {
                ...state,
                skills: action.skills,
            }
        }
        case GET_RESUME: {
            return {
                ...state,
                resume: action.resume,
            }
        }
        case GET_AVATAR: {
            return {
                ...state,
                avatar: action.avatar,
            }
        }
        case GET_BRIEF_PROFILE: {
            return {
                ...state,
                briefprofile: action.briefprofile,
                avatar: action.avatar
            }
        }
        default:
            return state
    }

}

export default ReducerProfile