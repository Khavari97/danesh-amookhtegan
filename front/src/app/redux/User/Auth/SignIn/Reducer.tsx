import {FetchDispatchTypes, INITIAL_STATE, IState} from "./ActionTypes";


const ReducerSignIn = (state = INITIAL_STATE, action: FetchDispatchTypes): IState => {
    switch (action.type) {
        case 'REQUEST': {
            return {
                ...state,
                loading: true
            }
        }

        case 'FETCH_SUCCESS': {
            return {
                ...state,
                loading: false,
                res: action.payload
            }
        }
        case 'FETCH_FAIL': {
            return {
                ...state,
                loading: false,
                error: action.payload
            }
        }
        default:
            return state
    }

}


export default ReducerSignIn