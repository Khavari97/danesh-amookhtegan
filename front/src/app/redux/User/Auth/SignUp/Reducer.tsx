import {INITIAL_STATE, IState, SET_PASSWORD, SET_PROFILE, SignUpTypes} from "./ActionTypes";


const ReducerSignup = (state = INITIAL_STATE, action: SignUpTypes): IState => {
    switch (action.type) {
        case SET_PASSWORD: {
            return {
                ...state,
                password: action.payload
            }
        }

        case SET_PROFILE: {
            return {
                ...state,
                profile: action.payload
            }
        }
        default:
            return state
    }

}


export default ReducerSignup