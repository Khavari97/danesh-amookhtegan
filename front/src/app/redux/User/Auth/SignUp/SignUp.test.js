import configureStore from 'redux-mock-store'
import { applyMiddleware, createStore } from 'redux'
import thunk from "redux-thunk";
import Reducer from "./Reducer";
import { actionsToStateSnapshot } from 'redux-state-snapshot'
import { INITIAL_STATE, IState, SET_PASSWORD, SET_PROFILE, SignUpTypes } from "./ActionTypes";


const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const mock_password = "1234567"

const setPassword = (password) => ({ type: SET_PASSWORD, payload: password })

it('should dispatch action', () => {

    // Initialize mockstore with empty state

    const createState = initialState => actions => actions.reduce(Reducer, initialState);
    const initialState = createState({
        password: '',
        profile: {},
    });
    const store = mockStore(initialState)

    // Dispatch the action
    store.dispatch(setPassword(mock_password))

    const actions = store.getActions()
    const state = store.getState()
    // Test if your store dispatched the expected actions
    expect(actions[0]).toEqual(setPassword(mock_password));
    // Test if your store dispatched the expected states

    expect(state).toEqual({
        password: mock_password,
        profile: {},
    });

})

it('to snapshot on failure', () => {
    const actionStateSnapshot = actionsToStateSnapshot([thunk])
    actionStateSnapshot(() => fetchDataFailure(), Reducer)
})
it('to snapshot on success', () => {
    const actionStateSnapshot = actionsToStateSnapshot()
    actionStateSnapshot(() => setPassword('123456'), Reducer)
})