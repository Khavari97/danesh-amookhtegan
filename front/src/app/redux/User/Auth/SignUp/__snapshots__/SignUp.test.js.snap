// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to snapshot on success: SET_PASSWORD 1`] = `
Object {
  "password": "123456",
  "profile": Object {},
}
`;
